src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -241,6 +241,22 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 
     register_cli_argument(scope, 'license_type', help="license type if the Windows image or disk used was licensed on-premises", **enum_choice_list(['Windows_Server', 'Windows_Client']))
 
+    register_cli_argument(scope, 'assign_identity', action='store_true', arg_group='Managed Service Identity',
+                          help='enables the VM/VMSS to autonomously, using its own managed identity, to directly authenticate and interact with other Azure services using bearer tokens')
+
+for scope in ['vm create', 'vmss create', 'vm assign-identity', 'vmss assign-identity']:
```

Comment:
It looks like you need to use the VersionConstraint object on the params that are only applicable to MSI, otherwise they will appear in the stack profile and crash when they are supplied.